projects/packages/forms/src/blocks/shared/hooks/use-find-block-recursively.js

```diff
@@ -0,0 +1,48 @@
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Custom hook to find a block recursively within the inner blocks of a given root block ID.
+ *
+ * @param {string}   searchRootClientId - The client ID of the block whose descendants will be searched.
+ * @param {Function} predicateFn        - A function that takes a block and returns true if it's the desired block.
+ * @return  {Object|null} The first block found that satisfies the predicate, or null if not found or if searchRootClientId is invalid.
+ */
+export const useFindBlockRecursively = ( searchRootClientId, predicateFn ) => {
+	const blocksToSearch = useSelect(
+		select => {
+			if ( ! searchRootClientId ) {
+				return [];
+			}
+			const { getBlocks } = select( blockEditorStore );
+			return getBlocks( searchRootClientId );
+		},
+		[ searchRootClientId ]
+	);
+
+	// Memoize the recursive search result to prevent re-computation on every render unless dependencies change
+	const foundBlock = useSelect( () => {
```

Comment:
I think it's a typo here:
```suggestion
	// Memoize the recursive search result to prevent re-computation on every render unless dependencies change
	const foundBlock = useMemo( () => {
```

Also, if `predicateFn` isn't using `useCallback` (as it isn't in later code), then it'll break this memoization.